File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        Set unique = new HashSet<Integer>();
        int k = 0;

        for (int n : nums) {
            if (!unique.contains(n)) {
                unique.add(n);
                nums[k] = n;
                k++;
            }
        }
        return k;
    }
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.ts
function removeDuplicates(nums: number[]): number {
    if (nums.length === 0) {
        return 0;
    }

    let r = 1;

    for (let i = 1; i < nums.length; i++) {
        if (nums[i] !== nums[i - 1]) {
            nums[r] = nums[i];
            r++;
        }
    }

    return r;
};